### ISSUE TITLE
Helm chart fails to create collector deployment and initContainer image is missing

### ISSUE THREAD

**Issue by FredrikAugust** [open-telemetry/opentelemetry-operator#2269]
### Component(s)

operator, collector, instrumentation

## Description

When syncing the helm chart for the operator (latest version: v0.41.0) it fails to create the collector deployment. I've attached logs from the operator — it appears the operator sees nothing wrong.

When I try to delete the `collector` CR from ArgoCD it starts a very strange loop where it will create/destroy sub-resources of the collector very rapidly. The `opentelemetry-opeartor` will create k8s events:

```
failed to create objects for collector: Operation cannot be fulfilled on deployments.apps "collector-collector": the object has been modified; please apply your changes to the latest version and try again
```

When deleting the collector resource manually from kubernetes using kubectl/Lens IDE and re-syncing it with ArgoCD it creates the deployment successfully.

This is only half the problem though, and I'm creating one issue as I assume they're related due to the fix being the same.

The second problem is that when I have the operator and collector running and create a new pod with the autoinstrumentation annotations it will fail to create the pod with the message:

```
Error creating: Pod "************" is invalid: spec.initContainers[1].image: Required value
```

This is once again fixed by deleting and recreating the `instrumentation` CR. After this it works as expected.

## Expected Result

The deployment should be created automatically, and instrumentation should inject correct image tag.

### Kubernetes Version

1.26.5-gke.2100

### Operator version

b6c75a

### Collector version

0.87.0

## Environment

- ArgoCD v2.8.4 for GitOps
- linkerd 2.14 for service mesh

### Log output

```shell
{"level":"info","ts":"2023-10-24T10:52:28Z","logger":"controllers.OpenTelemetryCollector","msg":"no upgrade routines are needed for the OpenTelemetry instance","name":"collector","namespace":"otel","version":"0.87.0","latest":"0.61.0"}
{"level":"info","ts":"2023-10-24T10:52:28Z","logger":"controllers.OpenTelemetryCollector","msg":"skipping upgrade for OpenTelemetry Collector instance","name":"collector","namespace":"otel"}
```

**Comment by FredrikAugust** [comment]
@jaronoff97 Hi! Sorry, forgot to respond. We upgraded to helm chart v0.41.0 which should correspond to v0.87.0

**Comment by jaronoff97** [comment]
You said that deleting and recreating the instrumentation CR made this work... can you re-iterate the current issue for me? I'm a bit lost with this one

**Comment by yuriolisa** [comment]
@FredrikAugust, did you have the chance to perform what @jaronoff97  suggested?

**Comment by FredrikAugust** [comment]
@yuriolisa Hi. I sadly don't remember what I did to fix it, but it works now. I don't know if it's still a problem, sadly.

**Comment by jaronoff97** [comment]
closing for now... please re-open if you see this again. Thanks!

**Comment by jaronoff97** [comment]
@hfranz-gebit this could because of a bug I fixed here #3074...  which is in release 0.108.0, if you upgrade to latest this shouldn't be an issue anymore (i hope!)

**Comment by hfranz-gebit** [comment]
@jaronoff97 Great! I'll test with the new release a few times in a dev environment!
